src/components/expanded-state/custom-gas/FeesPanel.js

```diff
@@ -0,0 +1,509 @@
+        navigate(Routes.EXPLAIN_SHEET, {
+          currentBaseFee,
+          currentGasTrend,
+          type,
+        });
+      };
+      let color;
+      let text;
+      if ((!error && !warning) || !selectedOptionIsCustom) {
+        color = isDarkMode
+          ? colors.alpha(darkModeThemeColors.blueGreyDark, 0.25)
+          : colors.alpha(colors.blueGreyDark, 0.25);
+        text = '􀅵';
+      } else if (error) {
+        color = colors.red;
+        text = '􀁟';
+      } else {
+        color = colors.yellowFavorite;
+        text = '􀇿';
+      }
+
+      return (
+        <PanelColumn>
+          <ButtonPressAnimation onPress={openGasHelper}>
+            <Row>
+              <PanelLabel>
+                {`${label} `}
+                <Label color={color} size="smedium" weight="bold">
+                  {text}
+                </Label>
+              </PanelLabel>
+            </Row>
+          </ButtonPressAnimation>
+        </PanelColumn>
+      );
+    },
+    [
+      currentBaseFee,
+      currentGasTrend,
+      navigate,
+      selectedOptionIsCustom,
+      isDarkMode,
+    ]
+  );
+
+  const formattedBaseFee = useMemo(
+    () => `${parseInt(currentBaseFee, 10)} Gwei`,
+    [currentBaseFee]
+  );
+
+  const handleOnInputFocus = useCallback(() => {
+    if (isEmpty(gasFeeParamsBySpeed[gasUtils.CUSTOM])) {
+      const gasFeeParams = gasFeeParamsBySpeed[selectedGasFeeOption];
+      updateToCustomGasFee({
+        ...gasFeeParams,
+        option: gasUtils.CUSTOM,
+      });
+    } else {
+      updateGasFeeOption(gasUtils.CUSTOM);
+    }
+  }, [
+    gasFeeParamsBySpeed,
+    selectedGasFeeOption,
+    updateGasFeeOption,
+    updateToCustomGasFee,
+  ]);
+
+  const handleFeesGweiInputFocus = useCallback(() => {
+    onCustomGasFocus?.();
+    handleOnInputFocus();
+    const {
+      gasFeeParams: { maxFeePerGas, maxPriorityFeePerGas },
+    } = selectedGasFee;
+    setCustomMaxPriorityFee(get(maxPriorityFeePerGas, 'gwei', 0));
+    setCustomMaxBaseFee(parseInt(get(maxFeePerGas, 'gwei', 0), 10));
+  }, [onCustomGasFocus, handleOnInputFocus, selectedGasFee]);
+
+  const handleCustomPriorityFeeFocus = useCallback(() => {
+    handleOnInputFocus();
+    setFeesGweiInputFocused(true);
+    handleFeesGweiInputFocus();
+  }, [handleFeesGweiInputFocus, handleOnInputFocus]);
+
+  const handleCustomPriorityFeeBlur = useCallback(() => {
+    setFeesGweiInputFocused(false);
+  }, [setFeesGweiInputFocused]);
+
+  const updateGasFee = useCallback(
+    ({ priorityFeePerGas = 0, feePerGas = 0 }) => {
+      const maxFeePerGas = selectedGasFee?.gasFeeParams?.maxFeePerGas;
+      const maxPriorityFeePerGas =
+        selectedGasFee?.gasFeeParams?.maxPriorityFeePerGas;
+
+      const gweiMaxPriorityFeePerGas = Number(maxPriorityFeePerGas?.gwei || 0);
+      const gweiMaxFeePerGas = Number(maxFeePerGas?.gwei || 0);
+
+      const newGweiMaxPriorityFeePerGas =
+        Math.round((gweiMaxPriorityFeePerGas + priorityFeePerGas) * 100) / 100;
```

Comment:
just dropping a note so we can just catch up on what the end goal is for this type of logic: rounding and * 100 / 100 etc (and if there's an easy utility function we can use in its place)